gnpy/core/equipment.py

```diff
@@ -0,0 +1,119 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+'''
+nf model parameters calculation
+calculate nf1, nf2 and Delta_P of a 2 coils edfa with internal VOA
+from nf_min and nf_max inputs 
+'''
+import numpy as np
+from gnpy.core.utils import lin2db, db2lin, load_json
+
+GAIN_MIN_FIELD = "gain_min"
+GAIN_MAX_FIELD = "gain_flatmax"
+NF_MIN_FIELD  ="nf_min"
+NF_MAX_FIELD = "nf_max"
+
+DEFAULT_CONFIG_JSON_FILENAME = 'default_edfa_config.json'
+ADVANCED_CONFIG_JSON_FILENAME = 'advanced_config_from_json'
+
+
+def nf_model(amp_dict):
+    gain_min = amp_dict[GAIN_MIN_FIELD]
+    gain_max = amp_dict[GAIN_MAX_FIELD]
+    try:
+        nf_min = amp_dict.get(NF_MIN_FIELD,-100)
+        nf_max = amp_dict.get(NF_MAX_FIELD,-100)
+        if nf_min<-10 or nf_max<-10:
+            raise ValueError
```

Comment:
The only thing in this block that could raise the `ValueError` is this explicit `raise`, but you handle it immediately below.

I think this can be simplified to:
```
if nf_min < -10 or nf_max < -10:
   print( ... )
```

Reply:
I want to stop the code execution.

Reply:
Okay, maybe we should remove the `except ValueError` so that the exception will terminate execution of this function.

Reply:
ok, will put the print in the if and remove the try/except bloc.